### PR TITLE
Add miscellaneous useful stats

### PR DESCRIPTION
**Pull request by jnwatson**
Add getBackupInfo() to cell about recent backup activity.
Add getSystemInfo() to cell about the running system.

Add run count, err count, recent errors to cron jobs and triggers.

### CODE REVIEW

File: synapse/lib/cell.py [hunk 1]
```diff
@@ -543,6 +566,25 @@ async def runBackup(self, name=None, wait=True):
...
 
+    @adminapi()
+    async def getBackupInfo(self):
+        '''
+        Get information about recent backup activity.
+
+        Returns:
+            (dict) It has the following keys:
+                - current_duration - If backup currently running, time in ms since backup started, otherwise None
+                - last_start - Last time (as ISO 8601 string) a backup started
```
Comment (invisig0th) [inline synapse/lib/cell.py:577]: These should be an epochmillis timestamps.

File: synapse/lib/layer.py [hunk 2]
```diff
@@ -1527,17 +1527,8 @@ async def getLayerSize(self):
...
-        for fpath, _, fnames in os.walk(self.dirn):
-            for fname in fnames:
-                fp = s_common.genpath(fpath, fname)
-                try:
-                    stat = os.stat(fp)
-                except OSError:  # pragma: no cover
-                    pass
-                else:
-                    totalsize += stat.st_blocks * 512
-        return totalsize
```
Comment (invisig0th) [inline synapse/lib/layer.py:1540]: nice mop up!

**Review by invisig0th — CHANGES_REQUESTED** [review]

File: synapse/lib/cell.py [hunk 3]
```diff
@@ -2231,3 +2345,51 @@ async def getCellInfo(self):
...
+            backupdisk = shutil.disk_usage(self.backdirn)
+            backupvolsize, backupvolfree = backupdisk.total, backupdisk.free
+        else:
+            backupvolsize, backupvolfree = 0, 0
+
+        myusage, myappusage = s_common.getDirSize(self.dirn)
+        totalmem = s_thisplat.getTotalMemory()
+        availmem = s_thisplat.getAvailableMemory()
+        pyversion = platform.python_version()
+
```
Comment (vEpiphyte) [inline synapse/lib/cell.py:2380]: cpu count should be free to get while we are here

File: synapse/lib/agenda.py [hunk 4]
```diff
@@ -328,6 +329,8 @@ def __init__(self, stor, iden, recur, indx, query, creator, recs, nexttime=None)
             self.nexttime = nexttime
         self.isrunning = False  # whether it is currently running
         self.startcount = 0  # how many times query has started
+        self.errcount = 0  # how many times this appt failed
```
[the remark this answers is not in the document]
Comment (jnwatson) [reply synapse/lib/agenda.py:332]: a) That's not what was in the spec
b) You can accidentally hide errors

File: synapse/lib/agenda.py [hunk 5]
```diff
@@ -391,11 +396,16 @@ def unpack(cls, stor, val):
         appt.doc = val.get('doc', '')
         appt.name = val.get('name', '')
         appt.startcount = val['startcount']
+        appt.errcount = val.get('errcount', 0)
```
[the remark this answers is not in the document]
Comment (jnwatson) [reply synapse/lib/agenda.py:399]: errcount is persistent for crons

File: synapse/lib/cell.py [hunk 6]
```diff
@@ -229,6 +231,27 @@ def getCellUser(self):
...
+        '''
+        Get info about the system in which the cell is running
+
+        Returns:
+            A dictionary with the following keys.  All size values are in bytes:
+                - volsize - Volume where cell is running total space
+                - volfree - Volume where cell is running free space
+                - backupvolsize - Backup directory volume total space
+                - backupvolfree - Backup directory volume free space
+                - celluptime - Cell uptime in milliseconds
```
Comment (vEpiphyte) [inline synapse/lib/cell.py:245]: the cell keys feel like they are better suited for ``getCellInfo``

**Review by invisig0th — APPROVED** [review]

**Pull request merged**